src/helpers/eventCache.ts

```diff
@@ -13,11 +13,9 @@ export class BaseEvents {
         this.instance.threads = new ThreadEvents(eventCache);
     }
 
-    private static instance: BaseEvents;
+    private static instance: BaseEvents = new BaseEvents();
+    
```

Comment:
@pwerry Thoughts on this approach?

Reply:
Better - doing this at class load will get rid of old-school singleton race conditions

###### <a href="https://getunblocked.com"><img src="https://avatars.githubusercontent.com/in/166219" width=20 align="center" /></a>&nbsp;&nbsp;Comment posted from [Unblocked](https://getunblocked.com/dashboard/team/461c33ed-2d8d-468d-9409-d2cec26e4741/thread/15710cfb-dd95-4c90-8170-a95d3b7adc6e?message=916b3a12-ea31-4e1e-871c-cd111a107164).

Reply:
Same idea!

###### <a href="https://getunblocked.com"><img src="https://avatars.githubusercontent.com/in/166219" width=20 align="center" /></a>&nbsp;&nbsp;Comment posted from [Unblocked](https://getunblocked.com/dashboard/team/461c33ed-2d8d-468d-9409-d2cec26e4741/thread/15710cfb-dd95-4c90-8170-a95d3b7adc6e?message=4a6f77a5-a9c1-4c60-9707-eb2aaf75fa87).

Reply:
Perfect!

###### <a href="https://getunblocked.com"><img src="https://avatars.githubusercontent.com/in/166219" width=20 align="center" /></a>&nbsp;&nbsp;Comment posted from [Unblocked](https://getunblocked.com/dashboard/team/461c33ed-2d8d-468d-9409-d2cec26e4741/thread/15710cfb-dd95-4c90-8170-a95d3b7adc6e?message=e2ffe6e7-26f3-4835-afb0-e944efdcf371).

Reply:
Let do it!

Reply:
Yaaay

###### <a href="https://getunblocked.com"><img src="https://avatars.githubusercontent.com/in/166219" width=20 align="center" /></a>&nbsp;&nbsp;Comment posted from [Unblocked](https://getunblocked.com/dashboard/team/461c33ed-2d8d-468d-9409-d2cec26e4741/thread/15710cfb-dd95-4c90-8170-a95d3b7adc6e?message=87f2925e-76ad-4640-b593-a6a8da5f2c77).

Reply:
Nah, changed my mind!

###### <a href="https://getunblocked.com"><img src="https://avatars.githubusercontent.com/in/166219" width=20 align="center" /></a>&nbsp;&nbsp;Comment posted from [Unblocked](https://getunblocked.com/dashboard/team/461c33ed-2d8d-468d-9409-d2cec26e4741/thread/15710cfb-dd95-4c90-8170-a95d3b7adc6e?message=a7818bcc-a15e-41e7-9f36-54f0d4b2dcc0).

Reply:
No, lets do it!

Reply:
My head is spinning!

###### <a href="https://getunblocked.com"><img src="https://avatars.githubusercontent.com/in/166219" width=20 align="center" /></a>&nbsp;&nbsp;Comment posted from [Unblocked](https://getunblocked.com/dashboard/team/461c33ed-2d8d-468d-9409-d2cec26e4741/thread/15710cfb-dd95-4c90-8170-a95d3b7adc6e?message=71fb35fc-6d97-4ad3-8906-5fbf3fc930b6).

Reply:
Here's a demo for Charlie!

Reply:
Hi Charlie!

--

Comment posted using [Unblocked](https://getunblocked.com/dashboard/team/461c33ed-2d8d-468d-9409-d2cec26e4741/thread/15710cfb-dd95-4c90-8170-a95d3b7adc6e?message=6c7e1aa3-5beb-4fed-9b3e-fba600040efe).